allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -269,6 +339,57 @@ def forward(self,  # type: ignore
             # TODO(matt): compute accuracy here.
             return outputs
 
+    def _compute_average_with_encoder(self, embed_tensor: Variable,
+                                      mask: Variable) -> Variable:
+        # computes the average along dimension 1 of the embed_tensor
+        dims = embed_tensor.size()
+        table_mask = mask.view(-1, mask.size(2))
+        averaged_tensor = self._entity_encoder(embed_tensor.view(-1, dims[2], dims[3]),
+                                               table_mask)
+        # (batch_size, num_entities, embedding_dim)
+        return averaged_tensor.view(dims[0], dims[1], dims[3])
+
+    def _get_neighbor_indexes(self, world: List[WikiTablesWorld],
+                                    max_num_entities: int) -> Variable:
+        # returns each entity's neighbor indexes
+        batches = []
+        for w in world:
+            # each batch has it's own world aka table
+            entities = w.table_graph.entities
+            entity2idx = {e: i for i, e in enumerate(entities)}
+            neighbors = w.table_graph.neighbors
+            neighbor_indexes = []
+            for e in entities:
+                # get indexes of entity e's neighbors
+                curr_neighbors = [entity2idx[cn] for cn in neighbors[e]]
+                # pad to length of max number of entities in a batch
+                padded = pad_sequence_to_length(curr_neighbors, max_num_entities)
+                neighbor_indexes.append(padded)
+            # pad the list with empty lists to hit length max num entities
```

Comment:
Because the name of the function is already pretty descriptive, this comment doesn't add much.  If you want a comment here, you could say something about _why_ we do this, e.g., because we're constructing a tensor later, but I'm not sure that's really necessary, either.  Same goes for a bunch of the comments in here - in general, [avoid comments that just say _what_ the code is doing](https://google.github.io/styleguide/cppguide.html#Implementation_Comments), unless it's not obvious.  Helpful comments give more context or explain why a particular decision was made.